conan/internal/api/detect/detect_api.py

```diff
@@ -488,11 +488,10 @@ def _cc_compiler(compiler_exe="cc"):
         compiler = "clang" if "clang" in out else "gcc"
         # clang and gcc have version after a space, first try to find that to skip extra numbers
         # that might appear in the first line of the output before the version
-        installed_version = re.search(r" ([0-9]+(\.[0-9]+)+)", out)
-        # Try only major but with spaces next
-        installed_version = installed_version or re.search(r" ([0-9]+(\.[0-9]+)?)", out)
+        # There might also be a leading parenthesis that contains build information, so we try to skip it
+        installed_version = re.search(r"(?:\(.*\))? ([0-9]+(\.[0-9]+)*)", out)
```

Comment:
There are 2 unrelated changes here., the first one is the commented parenthesis, but the other subtle one is the change from ``?`` (0 or one) -> ``*`` (0 or many). Why this second change? What is it fixing? Doesn't seem covered by tests.